src/lib.rs

```diff
@@ -525,10 +527,37 @@ fn run_single_on_query_match_callback<'a, 'b, 'c>(
     }
 }
 
-// pub struct RunForSliceStatus {
-//     pub violations: Vec<ViolationWithContext>,
-//     pub from_file_run_context_instance_provider: Box<dyn FromFileRunContextInstanceProvider>,
-// }
+#[self_referencing]
+pub struct PerConfigContext {
+    instantiated_rules: Vec<InstantiatedRule>,
+    #[borrows(instantiated_rules)]
+    #[covariant]
+    aggregated_queries: AggregatedQueries<'this>,
+}
+
+impl Clone for PerConfigContext {
```

Comment:
This is not really "intended" to get cloned, it was just required by `Cow` (which is "approximately" the borrowed vs owned abstraction I wanted)